src/main/java/it/gov/pagopa/reward/notification/service/iban/outcome/recovery/CompletedKoDiscardedRewardNotificationServiceImpl.java

```diff
@@ -0,0 +1,125 @@
+                                rewardIban.getUserId(),
+                                rewardIban.getInitiativeId(),
+                                RewardNotificationStatus.COMPLETED_KO)
+                        .flatMap(this::createRemedialNotification)
+                        .collectList(),
+                recovered -> "Recovered %d %s rewardNotification on userId %s and initiativeId %s".formatted(
+                        recovered.size(),
+                        RewardNotificationStatus.COMPLETED_KO,
+                        rewardIban.getUserId(),
+                        rewardIban.getInitiativeId()));
+    }
+
+    private Mono<RewardsNotification> createRemedialNotification(RewardsNotification discarded) {
+        log.info("[REWARD_NOTIFICATION_IBAN_OUTCOME] Found discarded {}} rewardNotification having id {} on userId {} and initiativeId {}",
+                RewardNotificationStatus.COMPLETED_KO, discarded.getId(), discarded.getUserId(), discarded.getInitiativeId());
+
+        return Mono.just(discarded)
+                .flatMap(this::buildRemedialNotification)
+                .flatMap(remedialNotification -> updateDiscardedAndStoreRemedial(discarded, remedialNotification))
+
+                .onErrorResume(e -> {
+                    log.error("[REWARD_NOTIFICATION_IBAN_OUTCOME] Something went wrong while recovering {} rewardNotification having id {} related to userId {} and initiativeId {}",
+                            RewardNotificationStatus.COMPLETED_KO, discarded.getId(), discarded.getUserId(), discarded.getInitiativeId(), e);
+                    return Mono.empty();
+                });
+    }
+
+    private Mono<RewardsNotification> buildRemedialNotification(RewardsNotification discarded) {
+
+        RewardsNotification remedial = discarded.toBuilder()
+                .status(RewardNotificationStatus.TO_SEND)
+                .exportId(null)
+                .exportDate(null)
+                .iban(null)
+                .rejectionReason(null)
+                .resultCode(null)
+                .feedbackDate(null)
+                .feedbackHistory(Collections.emptyList())
+                .cro(null)
+                .executionDate(null)
+                .remedialId(null)
+                .build();
+
+        setNewIdAndOrdinaryId(remedial, discarded);
+
+
+        return setRemedialNotificationDate(discarded.getInitiativeId(), remedial);
+    }
+
+    private void setNewIdAndOrdinaryId(RewardsNotification out, RewardsNotification input) {
+        String id = input.getId();
+
+        // if recovering a remedial it will calculate the next progressive
+        if (input.getOrdinaryId() != null) {
+            int nextRecoveryProgressiveId = getNextRecoveryProgressiveId(id);
+
+            out.setId("%s%s%d".formatted(input.getOrdinaryId(), RECOVERY_ID_SUFFIX, nextRecoveryProgressiveId));
+            out.setExternalId("%s%s%d".formatted(input.getExternalId(), RECOVERY_ID_SUFFIX, nextRecoveryProgressiveId));
```

Comment:
you are not removing the previous suffix:
thus first recovery would be -recovery-1, the next -recovery-1-recovery2 and so on...

ps: you could re-use the getNextRecoveryProgressiveId providing the externalId